gallery_dl/extractor/subscribestar.py

```diff
@@ -84,44 +86,98 @@ def _login_impl(self, username, password):
+                self._process_attachment_item,
+            ),
+            "link": ('data-href="', '"', self._process_media_item),
+            "audio": ('<source src="', '" type="audio/',
+                      self._process_media_item),
+        }
 
-        gallery = text.extr(html, 'data-gallery="', '"')
-        if gallery:
-            for item in util.json_loads(text.unescape(gallery)):
-                if "/previews" in item["url"]:
-                    self._warn_preview()
-                else:
-                    media.append(item)
-
-        attachments = text.extr(
-            html, 'class="uploads-docs"', 'data-role="post-edit_form"')
-        if attachments:
-            for att in attachments.split('class="doc_preview"')[1:]:
-                media.append({
-                    "id"  : text.parse_int(text.extr(
-                        att, 'data-upload-id="', '"')),
-                    "name": text.unescape(text.extr(
-                        att, 'doc_preview-title">', '<')),
-                    "url" : text.unescape(text.extr(att, 'href="', '"')),
-                    "type": "attachment",
-                })
+        for key, config in media_types.items():
+            if key in media_config:
+                start, end, processor = media_config[key]
+                segments = (
+                    text.extract_all(
+                        html,
+                        ((key, start, end),),
+                    )[0],
+                )
+                for segment in segments:
+                    if segment[key]:
+                        content = processor(segment, key)
+                        if content:
+                            media.append(content)
+        return media
+
+    def _process_gallery_item(self, item, media_type):
+        gallery_list = util.json_loads(text.unescape(item["gallery"]))
+        for media in gallery_list:
+            if "/previews" in media["url"]:
+                self._warn_preview()
+            return {"url": media["url"], "type": media_type}
+
+    def _process_attachment_item(self, item, media_type):
+        return {
+            "id": text.parse_int(text.extr(item, 'data-upload-id="', '"')),
+            "name": text.unescape(text.extr(item, 'doc_preview-title">', "<")),
+            "url": text.unescape(text.extr(item, 'href="', '"')),
+            "type": media_type,
+        }
+
+    def _process_media_item(self, item, media_type):
+        if media_type == "link" and util.check_if_supported_by_ytdlp(
+                item[media_type]):
+            return {"url": "ytdl:" + item[media_type], "type": media_type}
+        elif media_type == "audio":
+            return {"url": item[media_type], "type": media_type}
 
+    def _media_from_post(self, html):
+        media_types = {
+            "gallery": True,
+            "attachments": True,
+            "link": True,
+            "audio": True,
+        }
+        media = self._extract_media(html, media_types)
```

Comment:
We have rewritten the function to split it in multiple chunks to handle each media type, instead of having a big one.